etcdserver/api/etcdhttp/metrics.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdhttp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/coreos/etcd/etcdserver"
+	"github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/coreos/etcd/raft"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	pathMetrics = "/metrics"
+	pathHealth  = "/health"
+)
+
+// HandleMetricsHealth registers metrics and health handlers,
+// where '/metrics' serves health with prometheus metrics.
+func HandleMetricsHealth(mux *http.ServeMux, srv *etcdserver.EtcdServer) {
+	mux.Handle(pathMetrics, newMetricsHandler(srv))
+	mux.Handle(pathHealth, newHealthHandler(srv, "json"))
+}
+
+// HandlePrometheus registers prometheus handler on '/metrics'.
+func HandlePrometheus(mux *http.ServeMux) {
+	mux.Handle(pathMetrics, prometheus.Handler())
+}
+
+// HandleHealth registers health handler on '/health'.
+func HandleHealth(mux *http.ServeMux, srv *etcdserver.EtcdServer) {
+	mux.Handle(pathHealth, newHealthHandler(srv, "json"))
+}
+
+// newMetricsHandler serves health and metrics information.
+func newMetricsHandler(srv *etcdserver.EtcdServer) http.HandlerFunc {
+	h := prometheus.Handler()
+	hh := newHealthHandler(srv, "prometheus") // to be consistent with Prometheus handler
+	return func(w http.ResponseWriter, r *http.Request) {
+		hh(w, r)
```

Comment:
why is the handler still writing out the health if it can be accessed through `/health`? the metrics should probably match an ordinary client endpoint if possible...